packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -267,10 +267,13 @@ function useBlockEditorSettings( settings, postType, postId, renderingMode ) {
 			keepCaretInsideBlock,
 			mediaUpload: hasUploadPermissions ? mediaUpload : undefined,
 			__experimentalBlockPatterns: blockPatterns,
-			[ unlock( privateApis ).selectBlockPatternsKey ]: ( select ) =>
-				unlock( select( coreStore ) ).getBlockPatternsForPostType(
-					postType
-				),
+			[ unlock( privateApis ).selectBlockPatternsKey ]: ( select ) => {
+				const { isResolving, getBlockPatternsForPostType } = unlock(
+					select( coreStore )
+				);
+				const patterns = getBlockPatternsForPostType( postType );
+				return isResolving( 'getBlockPatterns' ) ? undefined : patterns;
```

Comment:
@jsnajdr This is a workaround to get the resolving state across stores. Not great, but it's a private setting for now. If it returns undefined, it means the data is still loading.

Reply:
Also even within the same store `isResolving` on `getBlockPatternsForPostType` doesn't work because it's not a resolver itself.

Anyway, curious if you have thoughts.